SAMtemplates/lambda_resources.yaml

```diff
@@ -0,0 +1,164 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Transform: AWS::Serverless-2016-10-31
+
+Description: >
+  Resources for a lambda
+
+Parameters:
+  CloudWatchKMSKey:
+    Type: String
+    Description: Cloudwatch KMS key
+    Default: none
+  SplunkSubscriptionFilterRole:
+    Type: String
+    Description: Subscription filter role for sending logs to splunk
+    Default: none
+  SplunkDeliveryStream:
+    Type: String
+    Description: Splunk delivery stream
+    Default: none
+  EnableSplunk:
+    Type: String
+    Description: Whether to use splunk
+    Default: false
+    AllowedValues: [true, false]
+  LambdaName:
+    Type: String
+    Description: Name of lambda we are creating for
+  TableName:
+    Type: String
+    Description: Name of the DynamoDB table
+  AccessToSpine:
+    Type: String
+    Description: Whether to use splunk
+    Default: false
+    AllowedValues: [true, false]
+  LogRetentionDays:
+    Type: Number
+    Description: How long to keep logs for
+    AllowedValues:
+      [
+        1,
+        3,
+        5,
+        7,
+        14,
+        30,
+        60,
+        90,
+        120,
+        150,
+        180,
+        365,
+        400,
+        545,
+        731,
+        1096,
+        1827,
+        2192,
+        2557,
+        2922,
+        3288,
+        3653,
+      ]
+
+Conditions:
+  ShouldUseSplunk: !Equals [true, !Ref EnableSplunk]
+  AllowAccessToSpine: !Equals [true, !Ref AccessToSpine]
+
+Resources:
+  LambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+            - sts:AssumeRole
+      ManagedPolicyArns:
+        - !ImportValue lambda-resources:LambdaInsightsLogGroupPolicy
+        - !ImportValue account-resources:LambdaEncryptCloudwatchKMSPolicy
+        - !ImportValue account-resources:LambdaDecryptSecretsKMSPolicy
+        - !Ref DynamoDBReadPolicy
+        - !Ref DynamoDBWritePolicy
```

Comment:
The attachment of these policies will change as per comments further down, but this lambda only puts items so should really only have the write policy. Where possible we should only be requiring exactly the permissions the lambda/resource needs.

Reply:
The DynamoDBReadPolicy has been removed.